### DISCUSSION TITLE
Cleanup of volumes and images

### DISCUSSION THREAD

**Discussion by hverlin** [firecow/gitlab-ci-local#872] [General]
Hi, 
Do we expect the users to clean up the volumes, images, etc. over time?

I have a lot of `gcl-*-tmp` and `gcl-*-build` volumes (taking 100+GB) on my system after using the tool a few times. 
There is some code to clean up the volumes https://github.com/firecow/gitlab-ci-local/blob/2c088f28e51f190cf08587397a9d74bfb8b3046d/src/job.ts#L527-L535 here but not sure if it always runs properly when [cancelling the command](https://github.com/firecow/gitlab-ci-local/blob/2c088f28e51f190cf08587397a9d74bfb8b3046d/src/handler.ts#L56-L56)

I was thinking that maybe that it could show a warning if some unused volumes are still present at the very end of the run? Or add some commands in the README like suggesting running 
```shell
docker volume rm $(docker volume ls -q --filter='name=gcl-')
```
from time to time?

**Comment by hverlin** [comment]
Maybe related https://github.com/firecow/gitlab-ci-local/issues/861#issuecomment-1559006765

**Comment by firecow** [comment]
Most certainly not.

More or less everything should be cleaned.

Maybe I've introduced a bug.

I'll test, when I get to my laptop.

**Comment by firecow** [comment]
https://github.com/firecow/gitlab-ci-local/pull/874

@hverlin I think i found my mistake, LOL :laughing:

**Reply by hverlin** [reply]
Thank you! Was not sure if it was a bug or expected behavior as it might not be always possible to clean things properly
